include/swift/AST/Expr.h

```diff
@@ -3706,15 +3702,15 @@ class ClosureExpr : public AbstractClosureExpr {
   SourceLoc InLoc;
 
   /// The explicitly-specified result type.
-  TypeLoc ExplicitResultType;
+  TypeExpr *ExplicitResultType;
```

Comment:
The only out of place thing about this is that `TypeExpr *` type is always a metatype which is not really intended here right? Maybe we should hide the fact that inside closure explicit type is represented via `TypeExpr` and instead only expose `TypeRepr` and `Type` to the outside?